src/host_capabilities/kubernetes.rs

```diff
@@ -0,0 +1,86 @@
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+
+/// Describe the set of parameters used by the `list_resources_by_namespace`
+/// function.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ListResourcesByNamespaceRequest {
+    /// apiVersion of the resource (v1 for core group, groupName/groupVersions for other).
+    pub api_version: String,
+    /// Singular PascalCase name of the resource
+    pub kind: String,
+    /// Namespace scoping the search
+    pub namespace: String,
+    /// A selector to restrict the list of returned objects by their labels.
+    /// Defaults to everything if `None`
+    pub label_selector: Option<String>,
+    /// A selector to restrict the list of returned objects by their fields.
+    /// Defaults to everything if `None`
+    pub field_selector: Option<String>,
+}
+
+/// Get all the Kubernetes resources defined inside of the given
+/// namespace
+/// Note: cannot be used with cluster-wide resources
+pub fn list_resources_by_namespace<T>(
+    req: &ListResourcesByNamespaceRequest,
+) -> Result<k8s_openapi::List<T>>
+where
+    T: k8s_openapi::ListableResource + serde::de::DeserializeOwned + Clone,
+{
+    let msg = serde_json::to_vec(req).map_err(|e| {
+        anyhow!(
+            "error serializing the list resources by namespace request: {}",
+            e
+        )
+    })?;
+    let response_raw = wapc_guest::host_call(
+        "kubewarden",
+        "kubernetes",
+        "list_resources_by_namespace",
+        &msg,
+    )
+    .map_err(|e| anyhow!("{}", e))?;
+
+    serde_json::from_slice(&response_raw).map_err(|e| {
+        anyhow!(
+            "error deserializing list resources by namespace response into Kubernetes resource: {:?}",
+            e
+        )
+    })
+}
+
+/// Describe the set of parameters used by the `list_all_resources` function.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ListAllResourcesRequest {
+    /// apiVersion of the resource (v1 for core group, groupName/groupVersions for other).
+    pub api_version: String,
+    /// Singular PascalCase name of the resource
+    pub kind: String,
+    /// A selector to restrict the list of returned objects by their labels.
+    /// Defaults to everything if `None`
+    pub label_selector: Option<String>,
+    /// A selector to restrict the list of returned objects by their fields.
+    /// Defaults to everything if `None`
+    pub field_selector: Option<String>,
+}
+
+/// Get all the Kubernetes resources defined inside of the cluster.
+/// Note: this has be used with cluster-wide resources
```

Comment:
Nit: typo here. IIUC, maybe better reworded to "Returns only cluster-wide resources"? (similar for `list_resources_by_namespace()`)

Reply:
No, this can be used to get all the pods declared inside of the cluster, not just the ones defined inside of namespace `foo`.

Moreover, this is the function you have to use to fetch cluster-wide resources, like `Namespace`.

Does it clarify?

Reply:
Yep! Still, typo, it should be "Note: this has to be used for cluster-wide resources" or similar.